File: Day294/predictTheColumn.go
package main

import (
	"fmt"
	"sort"
)

func columnWithMaxZeros(arr [][]int, N int) int {
	temp := make([][2]int, N)

	for i := 0; i < N; i++ {
		for j := 0; j < N; j++ {
			if arr[i][j] == 0 {
				temp[j][0] = j
				temp[j][1]++
			}
		}
	}

	sort.Slice(temp, func(i, j int) bool {
		if temp[i][1] != temp[j][1] {
			return temp[i][1] > temp[j][1]
		}
		return temp[i][0] < temp[j][0]
	})

	if temp[0][1] == 0 {
		return -1
	}

	return temp[0][0]
}

func main() {
	var t, N int
	fmt.Scan(&t)

	for t > 0 {
		fmt.Scan(&N)
		arr := make([][]int, N)
		for i := 0; i < N; i++ {
			arr[i] = make([]int, N)
			for j := 0; j < N; j++ {
				fmt.Scan(&arr[i][j])
			}
		}

		result := columnWithMaxZeros(arr, N)
		fmt.Println(result)

		t--
	}
}

File: Day294/predictTheColumn.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        
        while(t-- > 0)
        {
            int N = Integer.parseInt(read.readLine());
            int arr[][] = new int[N][N];
            
            for(int i = 0; i < N; i++)
            {
                int k = 0;
                String str[] = read.readLine().trim().split("\\s+");
                for(int j = 0; j < N; j++){
                  arr[i][j] = Integer.parseInt(str[k]);
                  k++;
                }
            }
            
            System.out.println(new Solution().columnWithMaxZeros(arr,N));
        }
    }
}

// } Driver Code Ends


//User function Template for Java

class Solution
{
    int columnWithMaxZeros(int arr[][], int N)
    {
        // code here 
        int temp[][] = new int[N][2];
        for (int i=0; i<N; i++) {
            for (int j=0; j<N; j++) {
                if (arr[i][j] == 0) {
                    temp[j][0] = j;
                    temp[j][1]++;
                }
            }
            
        }
        Arrays.sort(temp, (a,b)-> {
            if (a[0] != b[0] ){
                return b[1] - a[1];
            }
            return a[0] - b[0];
        });
        return temp[0][1] == 0 ? -1 : temp[0][0];
        
    }
}